Added Grog recipe and drink

Myfharad opened the pull request:
#### Summary
Content "Adds Grog recipe and drink"

#### Purpose of change

I wanted to add Grog as a way to explore new roleplaying aspects of CDDA. It's a simple drink, but one that can be easily produced cheaply.

#### Describe the solution

You can take a few different types of alcohol (Rum, Whiskey, Brandy), and a pinch of sugar, and use them to "enhance" the flavor of water.

#### Describe alternatives you've considered

No alternatives. I just thought it would be a neat thing to have.

#### Testing

Tested it, it works. 

#### Additional context

I was primarily inspired by [Townsends ](https://www.youtube.com/watch?v=JB7pk7AfV2E&t=232s) and [Tasting History with Max Miller](https://www.youtube.com/watch?v=oPTdSMOQRnY&t=504s) both give significant histories and historiographies of what Grog was in the 17th and 18th century. Grog to me is something that could easily become a substitute drink in CDDA to those who look at the game with a more Roleplaying aspect. Yes the world is infinite, and theoretically can spawn Alcohol indefinitely as long as you explore, but those who play the game slower, more ranch style as I do where I prefer to build a cozy homestead, Grog can be that drink you stockpile in kegs and drink at the end of the day after tending to your crops.

Yes Grog is primarily thought of as made with Rum, but from what I was able to gather online, the United States had their own version of Grog which they preferred which just [substituted Rum for Whiskey ](https://www.marinersmuseum.org/2021/03/spirits-on-the-uss-monitor-a-daily-dose-of-grog/)

If anything needs to be adjusted I'd appreciate any help.

Myfharad committed:
Grog recipe added.json

Myfharad committed:
Grog drink added.json

Myfharad committed:
Added Grog to Nested.json

Myfharad committed:
Fixed dumb typo.json

Myfharad committed:
Update data/json/recipes/recipe_food.json

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Maddremor commented:
Personally, I think you could just use `hard_liquor_unmixed`, clean water, and `sugar_standard_half` to make grog. It's hardly a specific drink as such given that the ingredients match up to what booze was available at the time.

Myfharad commented:
> Personally, I think you could just use `hard_liquor_unmixed`, clean water, and `sugar_standard_half` to make grog. It's hardly a specific drink as such given that the ingredients match up to what booze was available at the time.

valid points, though I don't believe it was until more modern interpretations where different types of hard liquor was brought in. I've been trying to source out documentation if tequila or vodka was a primary component during that era (also Moonshine is part of hard_liquor_unmixed, not sure how great that type of grog would be), but most Naval histories that included grog I've found primarily use Rum, Whiskey, or Brandy.

I'm not against using hard_liquor_unmixed, but personally I do prefer grog's closer historical recipe. I'm happy for more feedback to what others would prefer!

Maddremor commented:
The historical "recipe" is to water down strong liquor. It's not a fancy or specific thing. In my part of the world the term is still used to mean a mix of booze and something else (3+ ingredients making a cocktail). I would be kind to the player and let them use a broader array of ingredients for the drink to serve the same function.

Holli-Git commented:
> I get that you’re wanting functionality and flexibility for the player, but that fundamentally changes the recipe it was originally based on. And while I understand the modernization of the term for grog, just because any alcohol mixed with something else doesn’t make it the type of Grog im trying to add, thats like saying any sandwich with meat is a hamburger, completely ignoring its identity and specifics

I am quite confident that if sailors had access to more types of liquor, they would also make grog out of it.

Myfharad commented:
> I am quite confident that if sailors had access to more types of liquor, they would also make grog out of it.

redefining a term based on a hypothetical is akin to assuming that every culture on earth would've make every dish if they had access to the ingredients for it. 

I'm only trying to explain that recipe for Grog has a historical context and regional identity, despite it's incredibly simplistic recipe and eventual modernization, It's not just a generalized name for diluted liquor.

Myfharad committed:
Fix spacing issue I think.json

Myfharad committed:
Maybe a fix for build checks.json

Still learning CDDA JSON stuff so if this doesn't fix the checks i'll try something else lol

Maddremor commented:
Then if you want to be authentic you should probably axe the sugar part.

Myfharad commented:
> Then if you want to be authentic you should probably axe the sugar part.

like I said, it's not just diluted liquor, in the links I referenced it included two videos and one US navy documentation, in all of them they include histories of sailors adding sugar to make it palatable (they also included Lime which helped with scurvy but cdda has no Lime fruit, nor does lime grow in the New England region)

Maleclypse commented on data/json/items/comestibles/alcohol.json:
```diff
@@ -863,6 +863,36 @@
+    "description": "An early 17th century drink eventually popularized by sailors as a easy ration that contained alcohol and often a dash of sugar.  Though it barely contains any alcohol, it's probably better tasting than water, probably.",
+    "price": "5 cent",
+    "price_postapoc": "15 cent",
+    "material": [ "alcohol", "junk" ],
```
```suggestion
    "material": [ "alcohol", "water" ],
```

Maleclypse replied:
Ignore this while I determine whether Junk is determined by the inclusion of sugar vs hyper processed foodstuffs.

Myfharad replied:
> Ignore this while I determine whether Junk is determined by the inclusion of sugar vs hyper processed foodstuffs.

no worries, let me know if anything else is needed or changed so that the merge is pushed through

PGR-14 commented:
> > Then if you want to be authentic you should probably axe the sugar part.
> 
> like I said, it's not just diluted liquor, in the links I referenced it included two videos and one US navy documentation, in all of them they include histories of sailors adding sugar to make it palatable (they also included Lime which helped with scurvy but cdda has no Lime fruit, nor does lime grow in the New England region)

Lemon/Citric Acid? That may work.

Myfharad commented:
> Lemon/Citric Acid? That may work.

a neat idea, but I didn't see anything other than lime and sugar in the original navy ration recipe, so I'd like to keep it faithful to it's seafaring roots. plus personally it may deject players from ever touching the recipe since I feel like I hardly see lemons. (I know we have lemonade as a recipe in cdda but it's recipe is based off of lemons) At least with Rum, Whiskey, Brandy and Sugar it stays closer and faithful to the recipe and is far more renewable.
